app/test/vesting_test.go

```diff
@@ -0,0 +1,498 @@
+		vBals := []banktypes.Balance{
+			{
+				Address: cva.GetAddress().String(),
+				Coins:   cva.GetOriginalVesting().Add(initBalanceForGasFee),
+			},
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	// Test account details correctness
+	acli := authtypes.NewQueryClient(cctx.GRPCClient)
+	res, err := acli.Account(context.Background(), &authtypes.QueryAccountRequest{
+		Address: accAddress,
+	})
+	assert.NoError(s.T(), err)
+
+	var qAcc vestingtypes.ContinuousVestingAccount
+	err = qAcc.Unmarshal(res.Account.Value)
+	assert.NoError(s.T(), err)
+
+	// Checking the queried account data
+	assert.Equal(s.T(), accAddress, qAcc.GetAddress().String(), "account address must match")
+	assert.EqualValues(s.T(), startTime.Unix(), qAcc.GetStartTime(), "start time must match")
+	assert.EqualValues(s.T(), endTime.Unix(), qAcc.GetEndTime(), "end time must match")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+		"original vesting must match")
+
+	/*--------*/
+
+	// Test available balance
+	for i := 0; i < 5; i++ {
+		queryTime := tmtime.Now()
+
+		balances, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+		assert.NoError(s.T(), err)
+
+		minExpectedSpendableBal := cva.GetVestedCoins(queryTime).Add(initBalanceForGasFee)
+
+		assert.LessOrEqual(s.T(),
+			balances.AmountOf(app.BondDenom).Int64(),
+			minExpectedSpendableBal.AmountOf(app.BondDenom).Int64(),
+		)
+
+		_, err = cctx.WaitForTimestamp(startTime.Add(5)) // Wait for a while
+		assert.NoError(s.T(), err)
+	}
+	/*--------*/
+
+	s.testTransferVestingAmount(cctx, accName, false)
+	s.testDelegation(cctx, accName)
+}
+
+func (s *VestingModuleTestSuite) testTransferVestingAmount(cctx testnode.Context, accName string, mustSucceed bool) {
+	randomAcc := s.unusedAccount()
+	msgSend := banktypes.NewMsgSend(
+		getAddress(accName, cctx.Keyring),
+		getAddress(randomAcc, cctx.Keyring),
+		sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance))), // try to transfer the locked amount
+	)
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+	resTx, err := testnode.SignAndBroadcastTx(encCfg, cctx.Context, accName, []sdk.Msg{msgSend}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+
+	if mustSucceed {
+		assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the transfer TX must succeed")
+	} else {
+		assert.EqualValues(s.T(), 5, resQ.TxResult.Code, "the transfer TX must fail")
+	}
+}
+
+func (s *VestingModuleTestSuite) testDelegation(cctx testnode.Context, accName string) {
```

Comment:
You could further test that vested accounts can claim rewards  for their delegations

Reply:
Good idea